src/openassetio-python/cmodule/src/errors/exceptionsBinding.cpp

```diff
@@ -63,6 +64,12 @@ void setPyException(const Exception &exception, const py::module_ &pyModule,
  * given `exception_ptr` is rethrown and propagates up the nested
  * try-catch blocks until it is caught.
  *
+ * If the exception is a `HybridException` type, i.e. was originally a
+ * Python exception that has propagated through C++ and is now
+ * propagating back out to Python, then rethrow the original Python
+ * error. Otherwise, construct a new Python exception from the C++
+ * exception.
```

Comment:
[Question] Just so i'm clear. A hybrid exception is _always_ a python exception, in that, it's the c++ representation of an exception that was originally thrown in python?

Reply:
Yes exactly

Reply:
Hmm, i wonder if it might be better named, like, `PythonExceptionCppWrapper` or something. I was under the impression until now that it could be either/or.

One for later anyhow.